media-libs/openimageio/openimageio-2.5.5.0-r1.ebuild

```diff
@@ -0,0 +1,235 @@
+		media-libs/libglvnd
+		!qt6? (
+			dev-qt/qtcore:5
+			dev-qt/qtgui:5
+			dev-qt/qtopengl:5
+			dev-qt/qtwidgets:5
+		)
+		qt6? (
+			dev-qt/qtbase:6[gui,widgets,opengl]
+		)
+	)
+	raw? ( media-libs/libraw:= )
+	truetype? ( media-libs/freetype:2= )
+"
+DEPEND="
+	${RDEPEND}
+"
+
+DOCS=(
+	CHANGES.md
+	CREDITS.md
+	README.md
+)
+
+PATCHES="
+	"${FILESDIR}/fix_simd_test.patch"
+	"${FILESDIR}/fix_avx512_round.patch"
+"
+
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	use dicom || rm -r "${S}/src/dicom.imageio/" || die
+	cmake_src_prepare
+	cmake_comment_add_subdirectory src/fonts
+
+	if use test ; then
+		mkdir -p "${BUILD_DIR}"/testsuite || die
+		mv "${WORKDIR}/OpenImageIO-images-${TEST_OIIO_IMAGE_COMMIT}" "${BUILD_DIR}"/testsuite/oiio-images || die
+		mv "${WORKDIR}/openexr-images-${TEST_OEXR_IMAGE_COMMIT}" "${BUILD_DIR}"/testsuite/openexr-images || die
+	fi
+}
+
+src_configure() {
+	# Build with SIMD support
+	local cpufeature
+	local mysimd=()
+	for cpufeature in "${CPU_FEATURES[@]}"; do
+		use "${cpufeature%:*}" && mysimd+=("${cpufeature#*:}")
+	done
+
+	# If no CPU SIMDs were used, completely disable them
+	[[ -z ${mysimd[*]} ]] && mysimd=("0")
+
+	# This is currently needed on arm64 to get the NEON SIMD wrapper to compile the code successfully
+	# Even if there are no SIMD features selected, it seems like the code will turn on NEON support if it is available.
+	use arm64 && append-flags -flax-vector-conversions
+
+	# When building with more aggressive optimization flags, GCC will add FMA
+    # (Fused Multiply Add) instructions that will slightly alter the floating
+    # point operation results. This leads to test faliures.
+	append-flags -ffp-contract=off
+
+	local mycmakeargs=(
+		-DCMAKE_CXX_STANDARD="17"
+		-DDOWNSTREAM_CXX_STANDARD="17"
+		"-DVERBOSE=ON"
+		"-DOIIO_BUILD_TOOLS=$(usex tools)"
+		"-DBUILD_TESTING=$(usex test)"
+		"-DOIIO_BUILD_TESTS=$(usex test)"
+		"-DOIIO_DOWNLOAD_MISSING_TESTDATA=OFF"
+		"-DINSTALL_FONTS=OFF"
+		"-DBUILD_DOCS=$(usex doc)"
+		"-DINSTALL_DOCS=$(usex doc)"
+		"-DSTOP_ON_WARNING=OFF"
+		"-DUSE_CCACHE=OFF"
+		"-DUSE_DCMTK=$(usex dicom)"
+		"-DUSE_EXTERNAL_PUGIXML=ON"
+		"-DUSE_NUKE=OFF" # not in Gentoo
+		"-DUSE_FFMPEG=$(usex ffmpeg)"
+		"-DUSE_GIF=$(usex gif)"
+		"-DUSE_OPENJPEG=$(usex jpeg2k)"
+		"-DUSE_OPENCV=$(usex opencv)"
+		"-DUSE_OPENVDB=$(usex openvdb)"
+		"-DUSE_PTEX=$(usex ptex)"
+		"-DUSE_PYTHON=$(usex python)"
+		"-DUSE_LIBRAW=$(usex raw)"
+		"-DUSE_FREETYPE=$(usex truetype)"
+		"-DUSE_SIMD=$(local IFS=','; echo "${mysimd[*]}")"
+	)
+
+	if use gui; then
+		mycmakeargs+=( -DENABLE_IV=ON -DUSE_OPENGL=ON -DUSE_QT=ON )
+		if ! use qt6; then
+			mycmakeargs+=( -DCMAKE_DISABLE_FIND_PACKAGE_Qt6=ON )
+		fi
+	else
+		mycmakeargs+=( -DENABLE_IV=OFF -DUSE_QT=OFF )
+	fi
+
+	if use python; then
+		mycmakeargs+=(
+			"-DPYTHON_VERSION=${EPYTHON#python}"
+			"-DPYTHON_SITE_DIR=$(python_get_sitedir)"
+		)
+	fi
+
+	cmake_src_configure
+}
+
+src_test() {
+	tests_to_skip=()
```

Comment:
`CMAKE_SKIP_TESTS` from `cmake.eclass` does the same if you specify the tests as e.g `^jpeg2000-broken$`

Reply:
You create extra code that someone has to read and understand just to avoid doing it in the test name string.

I'd probably do:
```
CMAKE_SKIP_TESTS=(
  # jpeg2k
  "^jpeg2000-broken$"
  # openvdb
  "^openvdb-broken$"
  "^openvdb.batch-broken$"
  "^texture-texture3d-broken$"
  "^texture-texture3d.batch-broken$"
  # ptex
  "^ptex-broken$"
  # raw
  "^raw-broken$"
)
```
Or really just skip all broken tests:
```
CMAKE_SKIP_TESTS=(
  '-broken$'
)
```

Reply:
I think you might misunderstand what the tests do.
They test if for example the jpeg2k support/lib is broken.

If you compile without jpeg2k support, then naturally the test will fail.
However if you compile with jpeg2k support it will pass.

I think unconditionally skipping these tests kinda defeats the purpose of running the test suite in the first place as then the maintainer will not be able to easily tell or get any indication if any of the optional libraries are working or not.